src/NBitcoin/Key.cs

```diff
@@ -95,6 +95,12 @@ public ECDSASignature Sign(uint256 hash)
             return this._ECKey.Sign(hash);
         }
 
+        public ECDSASignature SignData(byte[] messageBytes)
```

Comment:
Maybe "SignMessage" to be consistent with what you did in pubkeys

Reply:
Ah yeah of course. Yes I think that's good then or maybe SignMessageData or SignMessageBytes. Whatever you think, I'd be ok with SignData too.

Reply:
Cool, I'll make it `SignMessageBytes` and `VerifyMessage`